importexport/musicxml/importmxmlpass2.cpp

```diff
@@ -4087,6 +4090,38 @@ static void handleDisplayStep(ChordRest* cr, int step, int octave, const Fractio
             }
       }
 
+//---------------------------------------------------------
+//   handleSmallness
+//---------------------------------------------------------
+
+/**
+ * Handle the distinction between small notes and a small 
+ * chord, to ensure a chord with all small notes is small.
+ * This also handles the fact that a small note being added
+ * to a small chord should not itself be small.
+ * I.e. a chord is "small until proven otherwise".
+ */
+
+static void handleSmallness(bool cueOrSmall, Note* note, Chord* c)
```

Comment:
I think it makes sense to handle this logic directly in Chord class. Just because it might be useful not only for import functionality, but for the whole domain model

We can consider it in the further PRs

Reply:
I agree! Taking it a step further, we could make the _small attribute of a Chord fully dependent on its member Notes (i.e. Chord::setSmall() just sets all the Notes to small, and Chord::small() reads whether all the Notes are small; Chord doesn't actually store a _small flag of its own). Not sure if that would introduce any regressions though. Just something to consider!

Reply:
Better leave such a change for the master branch only I guess.